src/main/scala/com/lightbend/kafkalagexporter/GraphiteEndpointSink.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package com.lightbend.kafkalagexporter
+
+import com.lightbend.kafkalagexporter.MetricsSink._
+import com.lightbend.kafkalagexporter.EndpointSink.ClusterGlobalLabels
+import java.net.Socket
+import java.io.PrintWriter
+import scala.util.{Try, Success, Failure}
+
+import scala.util.Try
+
+object GraphiteEndpointSink {
+
+  def apply(metricWhitelist: List[String], clusterGlobalLabels: ClusterGlobalLabels,
+            graphiteConfig: Option[GraphiteConfig]): MetricsSink = {
+    Try(new GraphiteEndpointSink(metricWhitelist, clusterGlobalLabels, graphiteConfig))
+      .fold(t => throw new Exception("Could not create Prometheus Endpoint", t), sink => sink)
+  }
+}
+
+class GraphiteEndpointSink private(metricWhitelist: List[String], clusterGlobalLabels: ClusterGlobalLabels,
+                                      graphiteConfig: Option[GraphiteConfig]) extends EndpointSink(clusterGlobalLabels) {
+  def graphitePush(graphiteConfig: GraphiteConfig, metricName: String, metricValue: Double): Unit = {
+    Try(new Socket(graphiteConfig.host, graphiteConfig.port)) match {
+      case Success(socket) =>
+        Try(new PrintWriter(socket.getOutputStream)) match {
+          case Success(writer) =>
+            writer.print(s"${graphiteConfig.prefix.getOrElse("")}${metricName} ${metricValue} ${System.currentTimeMillis / 1000}\n")
+            writer.close
+            socket.close
+          case Failure(_) =>
+            socket.close
+        }
+      case Failure(_) => {
+      }
+    }
+  }
+
+  def metricNameToGraphiteMetricName(metricValue: MetricValue): String = {
+    (getGlobalLabelValuesOrDefault(metricValue.clusterName) ++ metricValue.labels
+      ).map( x => x.replaceAll("\\.", "_")).mkString(".") + "." + metricValue.definition.name;
+  }
+
+  override def report(m: MetricValue): Unit = {
+    if (metricWhitelist.exists(m.definition.name.matches)) {
+      graphiteConfig.foreach { conf =>
+        graphitePush(conf, metricNameToGraphiteMetricName(m), m.value);
+      }
+    }
+  }
+
+  override def remove(m: RemoveMetric): Unit = {
```

Comment:
This is used in prometheus to unpublish a metric for a consumer group that disappears.  If a consumer group isn't reported for awhile and we don't push metrics for it to Graphite, what would the Graphite metric for that now removed consumed group look like?  Do we need to worry about unpublishing metrics?

Reply:
I don't think in graphite you have to unregister a metric since graphite is push based.
When the max retention is reached the metric will be totally removed by graphite itself.